### PR TITLE
Benchmarks tsne rfr

### PR DESCRIPTION
**Pull request by itearsl**

### CODE REVIEW

**Comment by Pahandrovich** [comment]
Can you rebase your branch?

**Review by KalyanovD — APPROVED** [review]

**Comment by KalyanovD** [comment]
Approved for merge after pep8 check and linux_sklearn steps fixes

File: cuml_bench/tsne.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# Load and convert data
+X, _, _, _ = bench.load_data(params)
+
+# Create our random forest regressor
+tsne = TSNE(n_components=params.n_components, early_exaggeration=params.early_exaggeration,
+            learning_rate=params.learning_rate, angle=params.angle,
+            min_grad_norm=params.min_grad_norm, random_state=params.random_state)
+
+fit_time, _ = bench.measure_function_time(tsne.fit, X, params=params)
+# divergence = tsne.kl_divergence_
```
Comment (KalyanovD) [inline cuml_bench/tsne.py:31]: need to remove, or make it meaningful, like "TODO: investigate how to compare skl and cuml benchs"

**Comment by outoftardis** [comment]
I'll have comments for the wording of argument descriptions, please wait for my review before merging

File: cuml_bench/tsne.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                    'original space are in the embedded space and how much '
+                    'space will be between them.')
+parser.add_argument('--learning-rate', type=float, default=200.0,
+                    help='The learning rate for t-SNE is usually in the range [10.0, 1000.0].')
+parser.add_argument('--angle', type=float, default=0.5,
+                    help='Only used if method=’barnes_hut’ This is the trade-off'
+                    'between speed and accuracy for Barnes-Hut T-SNE.')
+parser.add_argument('--min-grad-norm', type=float, default=1e-7,
+                    help='If the gradient norm is below this threshold,'
+                    'the optimization will be stopped.')
```
Comment (outoftardis) [inline cuml_bench/tsne.py:20]:
```suggestion
                    'the optimization is stopped.')
```

File: datasets/load_datasets.py [hunk 3]
```diff
@@ -1,106 +1,107 @@
...
+    else:
+        logging.warning(f"There is no script to download the dataset: {dataset_name}. "
+                        "You need to add a dataset or script to load it.")
+        return False
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Use \'-d\' or \'--datasets\' option to enumerate '
+                    'dataset(s) which should be downloaded')
```
Comment (outoftardis) [inline datasets/load_datasets.py:88]:
```suggestion
                    'dataset(s) that should be downloaded')
```

File: datasets/load_datasets.py [hunk 4]
```diff
@@ -1,106 +1,107 @@
...
+    parser.add_argument('-l', '--list', action='store_const',
+                        const=True, help='list of available datasets')
```
Comment (outoftardis) [inline datasets/load_datasets.py:90]:
```suggestion
                        const=True, help='The list of available datasets')
```

File: datasets/load_datasets.py [hunk 5]
```diff
@@ -1,106 +1,107 @@
...
+    parser.add_argument('-d', '--datasets', type=str, nargs='*',
+                        help='datasets which should be downloaded')
```
Comment (outoftardis) [inline datasets/load_datasets.py:92]:
```suggestion
                        help='The datasets that should be downloaded.')
```

File: datasets/load_datasets.py [hunk 6]
```diff
@@ -1,106 +1,107 @@
...
+        sys.exit(0)
+
+    root_dir = Path(os.environ['DATASETSROOT'])
+
+    if args.datasets is not None:
+        for val in dataset_loaders.values():
+            val(root_dir)
+    else:
+        logging.warning(
+            'Warning: Enumerate dataset(s) which should be downloaded')
```
Comment (outoftardis) [inline datasets/load_datasets.py:107]:
```suggestion
            'Warning: Enumerate dataset(s) that should be downloaded')
```

File: sklearn_bench/tsne.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+        alg_instance=tsne,
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='scikit-learn tsne '
+                                     'regression benchmark')
+
+    parser.add_argument('--n-components', type=int, default=2,
+                        help='Dimension of the embedded space.')
```
Comment (outoftardis) [inline sklearn_bench/tsne.py:54]:
```suggestion
                        help='The dimension of the embedded space.')
```

File: sklearn_bench/tsne.py [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    parser.add_argument('--early-exaggeration', type=float, default=12.0,
+                        help='Controls how tight natural clusters in the'
+                        'original space are in the embedded space and how '
+                        'much space will be between them.')
```
Comment (outoftardis) [inline sklearn_bench/tsne.py:58]: same comment for this description as in `cuml_bench/tsne.py`

File: sklearn_bench/tsne.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+    parser.add_argument('--learning-rate', type=float, default=200.0,
+                        help='The learning rate for t-SNE is usually in the range [10.0, 1000.0].')
+    parser.add_argument('--angle', type=float, default=0.5,
+                        help='Only used if method=’barnes_hut’'
+                        'This is the trade-off between speed and accuracy for Barnes-Hut T-SNE.')
```
Comment (outoftardis) [inline sklearn_bench/tsne.py:63]: same comment for this description as in `cuml_bench/tsne.py`

File: sklearn_bench/tsne.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+    parser.add_argument('--min-grad-norm', type=float, default=1e-7,
+                        help='If the gradient norm is below this threshold,'
+                        'the optimization will be stopped.')
```
Comment (outoftardis) [inline sklearn_bench/tsne.py:66]:
```suggestion
                        'the optimization is stopped.')
```

File: cuml_bench/tsne.py (hunk 1 above)
Comment (Pahandrovich) [reply cuml_bench/tsne.py:32]:
```suggestion
# TODO: investigate how to compare sklearn and cuml metrics for tsne
```

**Review by outoftardis — APPROVED** [review]
lgtm

**Review by Pahandrovich — APPROVED** [review]

**Pull request merged**